src/extension/src/ActionHandler.py

```diff
@@ -87,10 +87,14 @@ def tear_down(self):
     def setup_telemetry(self):
         # check if events folder exists, if it does init telemetry, if events folder does not exist, log that telemetry is not supported by agent since events folder does not exist
         events_folder = self.ext_env_handler.events_folder
-        if events_folder is None or not os.path.exists(events_folder):
+        if events_folder is None:
             err_msg = Constants.TELEMETRY_AT_AGENT_NOT_COMPATIBLE_ERROR_MSG
             self.logger.log_error(err_msg)
         else:
+            if not os.path.exists(events_folder):
```

Comment:
informational: elif: would have been functionally the same.

Reply:
Not sure if I understand what you mean here. if the else was replaced with elif, it would only perform the logic inside if the events folder didn't exist, instead of performing the logic whether it existed or not.

Reply:
_Re: it would only perform the logic inside if the events folder didn't exist, instead of performing the logic whether it existed or not._
-- Which is exactly the same as what's happening with line 95 and 96 now. If there were other statements outside of the 'if' block in what is line 94 in your PR, what you're concerned about would be true.

Reply:
Is it the ones on line 98 and 99? Which are in else but will not be in "if not os.path.exists(events_folder):"

Reply:
Ignore -- indentation was misunderstood for 98, 99.